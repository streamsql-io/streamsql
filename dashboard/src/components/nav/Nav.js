import React from "react";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import { makeStyles } from "@material-ui/core/styles";
import Drawer from "@material-ui/core/Drawer";
import Divider from "@material-ui/core/Divider";
import Box from "@material-ui/core/Box";
import { PURPLE } from "styles/theme";
import Grid from "@material-ui/core/Grid";
import List from "@material-ui/core/List";
import ListSubheader from "@material-ui/core/ListSubheader";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import { BrowserRouter as Router, NavLink } from "react-router-dom";

import Icon from "@material-ui/core/Icon";

const drawerWidth = "15%";

const useStyles = makeStyles((theme) => ({
  root: {
    height: "100vh",
    overflow: "hidden",
  },
  appBar: {
    position: "fixed",
    zIndex: theme.zIndex.drawer + 1,
    top: 0,
    left: 0,
    right: 0,
    overflow: "hidden",
  },
  drawer: {
    width: drawerWidth,
    flexShrink: 0,
  },
  drawerPaper: {
    width: drawerWidth,
  },
  toolbar: theme.mixins.toolbar,
  container: {
    height: "100vh",
    width: "100%",
    overflow: "hidden",
  },
  content: {
    height: "100vh",
    width: "100%",
    overflow: "hidden",
    flexWrap: "noWrap",
  },
  navlink_active: {
    color: PURPLE,
    // Prior to this, elements inside of <a> were overriding color, this
    // makes it more specific since we can't use !important.
    "& span": {
      color: PURPLE,
    },
  },
  main: {
    flexGrow: 1,
    backgroundColor: theme.palette.background.default,
    overflowY: "auto",
    "&::-webkit-scrollbar-track": {
      backgroundColor: "#FFFFFF",
    },

    "&::-webkit-scrollbar": {
      width: "12px",
      backgroundColor: "#FFFFFF",
    },

    "&::-webkit-scrollbar-thumb": {
      backgroundColor: "rgba(0, 0, 0, 0.2)",
    },
    "&::-webkit-scrollbar-button": {
      display: "none",
      width: 0,
      height: 0,
    },
  },
}));

const Nav = ({ children }) => {
  const classes = useStyles();
  return (
    <Router>
      <Box display="flex" component="div" className={classes.root}>
        <NavDrawer classes={classes} />
        <Box className={classes.container}>
          <TopBar classes={classes} />
          <BelowTopBar classes={classes}>
            <Box component="main" className={classes.main}>
              {children}
            </Box>
          </BelowTopBar>
        </Box>
      </Box>
    </Router>
  );
};

const TopBar = ({ classes }) => (
  <AppBar className={classes.appBar}>
    <Toolbar>
      <Typography component="span" variant="h5">
        StreamSQL
      </Typography>
    </Toolbar>
  </AppBar>
);

const BelowTopBar = ({ children, classes }) => (
  <Grid container direction="column" className={classes.content}>
    {/* Adds a div the size of TopBar to shift everything under it */}
    <div className={classes.toolbar} />
    {children}
  </Grid>
);

const NavDrawer = ({ classes }) => (
  <Drawer
    className={classes.drawer}
    variant="permanent"
    classes={{
      paper: classes.drawerPaper,
    }}
    anchor="left"
  >
    <div className={classes.toolbar} />
    <Divider />
    <ResourcesDrawerList classes={classes} />
    <Divider />
    <MonitoringDrawerList classes={classes} />
    <Divider />
    <AdminDrawerList classes={classes} />
  </Drawer>
);

const ResourcesDrawerList = ({ classes }) => {
  const items = [
    { text: "Data Sources", icon: "file-import", path: "/sources" },
    { text: "Materialized Views", icon: "copy", path: "/views" },
    { text: "Features", icon: "file-code", path: "/features" },
    { text: "Feature Sets", icon: "sitemap", path: "/feature-sets" },
    { text: "Training Sets", icon: "archive", path: "/training-sets" },
  ];
  return <DrawerList classes={classes} name="Resources" items={items} />;
};

const MonitoringDrawerList = ({ classes }) => {
  const items = [
    { text: "Metrics", icon: "chart-line", path: "/metrics" },
    { text: "Deployment", icon: "server", path: "/deployment" },
  ];
  return <DrawerList classes={classes} name="Monitoring" items={items} />;
};

const AdminDrawerList = ({ classes }) => {
  const items = [
    { text: "Users", icon: "users", path: "/users" },
    { text: "Settings", icon: "cogs", path: "/settings" },
    { text: "Billing", icon: "wallet", path: "/billing" },
    {
      text: "Documentation",
      icon: "book",
      path: "https://docs.streamsql.io",
      external: true,
    },
    { text: "Help", icon: "question", path: "/help" },
  ];
  return <DrawerList classes={classes} name="Administration" items={items} />;
};

export const DrawerList = ({ classes, name, items }) => (
  <List>
    <ListSubheader>{name}</ListSubheader>
    {items.map(({ text, icon, path, external }) => (
      <DrawerListLink
        key={text}
        path={path}
        classes={classes}
        external={external}
      >
        <ListItem button>
          <ListItemIcon>
            {/* Prior to overflow being set to visible, fa-sitemap was being
              cut-off since its slightly larger than a typical icon. */}
            <Icon style={{ overflow: "visible" }} className={`fa fa-${icon}`} />
          </ListItemIcon>
          <ListItemText primary={text} />
        </ListItem>
      </DrawerListLink>
    ))}
  </List>
);

export function DrawerListLink({ classes, path, external, children }) {
  if (external) {
    return (
      // _blank opens a new tab and noopender noreferrer blocks a known security
      // issue. Read more here: https://mathiasbynens.github.io/rel-noopener/
      <a target="_blank" rel="noopener noreferrer" href={path}>
        {children}
      </a>
    );
  } else {
    return (
      <NavLink to={path} activeClassName={classes.navlink_active}>
        {children}
      </NavLink>
    );
  }
}

export default Nav;
