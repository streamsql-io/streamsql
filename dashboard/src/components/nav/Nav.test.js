import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";

import { DrawerLists, DrawerList, DrawerListLink, Nav, TopBar } from "./Nav.js";

configure({ adapter: new Adapter() });

describe("Nav", () => {
  const classes = {
    navlink_active: "",
    appBar: "",
  };

  it("hasn't changed", () => {
    const tree = renderer
      .create(
        <Nav>
          <div>abc</div>
          <div>def</div>
        </Nav>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe("TopBar", () => {
    it("hasn't changed", () => {
      const tree = renderer.create(<TopBar classes={classes} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("DrawerLists", () => {
    const drawerLists = shallow(
      <DrawerLists classes={classes} sections={[{}, {}, {}]} />
    );

    it("renders sections", () => {
      expect(drawerLists.find("DrawerList")).toHaveLength(3);
    });

    it("prepends divider", () => {
      expect(drawerLists).toHaveLength(6);
      for (var i of [0, 2, 4]) {
        expect(drawerLists.at(i).name()).toMatch(/Divider/);
      }
    });
  });

  describe("DrawerList", () => {
    const items = [
      { text: "Users", icon: "users", path: "/users" },
      { text: "Settings", icon: "cogs", path: "/settings" },
      { text: "Billing", icon: "wallet", path: "/billing" },
      {
        text: "Documentation",
        icon: "book",
        path: "https://docs.streamsql.io",
        external: true,
      },
      { text: "Help", icon: "question", path: "/help" },
    ];
    const name = "Test List";
    const drawerList = shallow(
      <DrawerList classes={classes} name={name} items={items} />
    );
    it("renders header", () => {
      const header = drawerList.children().first();
      expect(header.text()).toEqual(name);
      // Material-UI wraps ListSubheader to add styles, so we use a regex.
      expect(header.name()).toMatch(/ListSubheader/);
    });

    it("renders items", () => {
      // Ignore the header
      const renderedItems = drawerList.children().slice(1);
      expect(renderedItems).toHaveLength(items.length);
      renderedItems.forEach((node) => {
        expect(node.name()).toEqual("DrawerListLink");
      });
    });
  });

  describe("DrawerListLink", () => {
    // These tests are run on both internal and external NavLinks
    const genericTests = (val) => {
      it("renders children", () => {
        expect(val.children()).toHaveLength(1);
        expect(val.first().text()).toEqual("abc");
      });
    };

    describe("internal", () => {
      const internal = shallow(
        <DrawerListLink path="/abc" classes={classes}>
          <div>abc</div>
        </DrawerListLink>
      );
      genericTests(internal);
      it("Sets path correctly", () => {
        expect(internal.props().to).toEqual("/abc");
      });
    });

    describe("external", () => {
      const external = shallow(
        <DrawerListLink
          path="https://streamsql.io"
          classes={classes}
          external={true}
        >
          <div>abc</div>
        </DrawerListLink>
      );
      genericTests(external);
      it("Sets href correctly", () => {
        expect(external.props().href).toEqual("https://streamsql.io");
      });
    });
  });
});
