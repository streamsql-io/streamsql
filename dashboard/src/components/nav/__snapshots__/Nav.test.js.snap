// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav TopBar hasn't changed 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <span
      className="MuiTypography-root MuiTypography-h5"
    >
      StreamSQL
    </span>
  </div>
</header>
`;

exports[`Nav hasn't changed 1`] = `
<div
  className="MuiBox-root MuiBox-root-10 makeStyles-root-1"
>
  <div
    className="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-3"
  >
    <div
      className="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-4 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
    >
      <div
        className="makeStyles-toolbar-5"
      />
      <hr
        className="MuiDivider-root"
      />
      <ul
        className="MuiList-root MuiList-padding"
      >
        <li
          className="MuiListSubheader-root MuiListSubheader-sticky MuiListSubheader-gutters"
        >
          Resources
        </li>
        <a
          aria-current={null}
          href="/sources"
          onClick={[Function]}
        >
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root"
            >
              <span
                aria-hidden={true}
                className="material-icons MuiIcon-root fa fa-file-import"
                style={
                  Object {
                    "overflow": "visible",
                  }
                }
              />
            </div>
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Data Sources
              </span>
            </div>
          </div>
        </a>
        <a
          aria-current={null}
          href="/views"
          onClick={[Function]}
        >
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root"
            >
              <span
                aria-hidden={true}
                className="material-icons MuiIcon-root fa fa-copy"
                style={
                  Object {
                    "overflow": "visible",
                  }
                }
              />
            </div>
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Materialized Views
              </span>
            </div>
          </div>
        </a>
        <a
          aria-current={null}
          href="/features"
          onClick={[Function]}
        >
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root"
            >
              <span
                aria-hidden={true}
                className="material-icons MuiIcon-root fa fa-file-code"
                style={
                  Object {
                    "overflow": "visible",
                  }
                }
              />
            </div>
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Features
              </span>
            </div>
          </div>
        </a>
        <a
          aria-current={null}
          href="/feature-sets"
          onClick={[Function]}
        >
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root"
            >
              <span
                aria-hidden={true}
                className="material-icons MuiIcon-root fa fa-sitemap"
                style={
                  Object {
                    "overflow": "visible",
                  }
                }
              />
            </div>
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Feature Sets
              </span>
            </div>
          </div>
        </a>
        <a
          aria-current={null}
          href="/training-sets"
          onClick={[Function]}
        >
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root"
            >
              <span
                aria-hidden={true}
                className="material-icons MuiIcon-root fa fa-archive"
                style={
                  Object {
                    "overflow": "visible",
                  }
                }
              />
            </div>
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Training Sets
              </span>
            </div>
          </div>
        </a>
      </ul>
      <hr
        className="MuiDivider-root"
      />
      <ul
        className="MuiList-root MuiList-padding"
      >
        <li
          className="MuiListSubheader-root MuiListSubheader-sticky MuiListSubheader-gutters"
        >
          Monitoring
        </li>
        <a
          aria-current={null}
          href="/metrics"
          onClick={[Function]}
        >
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root"
            >
              <span
                aria-hidden={true}
                className="material-icons MuiIcon-root fa fa-chart-line"
                style={
                  Object {
                    "overflow": "visible",
                  }
                }
              />
            </div>
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Metrics
              </span>
            </div>
          </div>
        </a>
        <a
          aria-current={null}
          href="/deployment"
          onClick={[Function]}
        >
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root"
            >
              <span
                aria-hidden={true}
                className="material-icons MuiIcon-root fa fa-server"
                style={
                  Object {
                    "overflow": "visible",
                  }
                }
              />
            </div>
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Deployment
              </span>
            </div>
          </div>
        </a>
      </ul>
      <hr
        className="MuiDivider-root"
      />
      <ul
        className="MuiList-root MuiList-padding"
      >
        <li
          className="MuiListSubheader-root MuiListSubheader-sticky MuiListSubheader-gutters"
        >
          Admin
        </li>
        <a
          aria-current={null}
          href="/users"
          onClick={[Function]}
        >
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root"
            >
              <span
                aria-hidden={true}
                className="material-icons MuiIcon-root fa fa-users"
                style={
                  Object {
                    "overflow": "visible",
                  }
                }
              />
            </div>
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Users
              </span>
            </div>
          </div>
        </a>
        <a
          aria-current={null}
          href="/settings"
          onClick={[Function]}
        >
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root"
            >
              <span
                aria-hidden={true}
                className="material-icons MuiIcon-root fa fa-cogs"
                style={
                  Object {
                    "overflow": "visible",
                  }
                }
              />
            </div>
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Settings
              </span>
            </div>
          </div>
        </a>
        <a
          aria-current={null}
          href="/billing"
          onClick={[Function]}
        >
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root"
            >
              <span
                aria-hidden={true}
                className="material-icons MuiIcon-root fa fa-wallet"
                style={
                  Object {
                    "overflow": "visible",
                  }
                }
              />
            </div>
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Billing
              </span>
            </div>
          </div>
        </a>
        <a
          href="https://docs.streamsql.io"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root"
            >
              <span
                aria-hidden={true}
                className="material-icons MuiIcon-root fa fa-book"
                style={
                  Object {
                    "overflow": "visible",
                  }
                }
              />
            </div>
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Documentation
              </span>
            </div>
          </div>
        </a>
        <a
          aria-current={null}
          href="/help"
          onClick={[Function]}
        >
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root"
            >
              <span
                aria-hidden={true}
                className="material-icons MuiIcon-root fa fa-question"
                style={
                  Object {
                    "overflow": "visible",
                  }
                }
              />
            </div>
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Help
              </span>
            </div>
          </div>
        </a>
      </ul>
    </div>
  </div>
  <div
    className="MuiBox-root MuiBox-root-11 makeStyles-container-6"
  >
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-2 mui-fixed MuiPaper-elevation4"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <span
          className="MuiTypography-root MuiTypography-h5"
        >
          StreamSQL
        </span>
      </div>
    </header>
    <div
      className="MuiGrid-root makeStyles-content-7 MuiGrid-container MuiGrid-direction-xs-column"
    >
      <div
        className="makeStyles-toolbar-5"
      />
      <main
        className="MuiBox-root MuiBox-root-12 makeStyles-main-9"
      >
        <div>
          abc
        </div>
        <div>
          def
        </div>
      </main>
    </div>
  </div>
</div>
`;
